java/memory/memory-core/src/main/java/org/apache/arrow/memory/AllocationManager.java

```diff
@@ -48,8 +49,9 @@ public abstract class AllocationManager {
   // This is mostly a semantic constraint on the API user: if the reference count reaches 0 in the owningLedger, then
   // there are not supposed to be any references through other allocators. In practice, this doesn't do anything
   // as the implementation just forces ownership to be transferred to one of the other extant references.
-  private volatile BufferLedger owningLedger;
+  private volatile @Nullable BufferLedger owningLedger;
```

Comment:
Needed to cover https://github.com/davisusanibar/arrow/blob/46dd019852529e8700cc6abe016203f24e3d498f/java/memory/memory-core/src/main/java/org/apache/arrow/memory/AllocationManager.java#L159

Reply:
I think there is a point here. It could be initially non-null but after calling the state changes to null. So we could determine it as nullable? @danepitkin what do you think?

Reply:
If there is one way to assign null, then the variable must be mapped as Nullable. It is possible to make some changes, but that is the rule.